compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -2567,10 +2567,13 @@ boolean checkTypeAndVarCountConsistency(BLangTupleVariable varNode, BTupleType t
                 case TypeTags.ARRAY:
                     List<BType> tupleTypes = new ArrayList<>();
                     BArrayType arrayType = (BArrayType) referredType;
+                    tupleTypeNode = new BTupleType(tupleTypes);
+                    if (varNode.restVariable != null) {
```

Comment:
Moving this if block after the for loop is more cleaner

Reply:
Updated. 6bbd11e31ea08006fe819b0957239a02cacc981b